src/map/mob.c

```diff
@@ -2757,8 +2757,8 @@ static int mob_dead(struct mob_data *md, struct block_list *src, int type)
 
 					drop_rate = (int)(0.5 + drop_rate * drop_rate_bonus / 100.);
 
-					// Limit drop rate, default: 90%
-					drop_rate = min(drop_rate, 9000);
+					// Make sure drop rate does not go beyond 100% after the drop rate modifiers are applied
+					drop_rate = min(drop_rate, 10000);
```

Comment:
I am not sure why previous developers decided to use cap rate of 90%, but one thing is better to de-hardcode the value 90% into battle-conf.

Reply:
I think the 90% cap is the official behavior: https://irowiki.org/wiki/Drop_System
> There is a cap of 90% drop chance on items boosted by anything except database edits. That means if an item has a 50% chance to drop and it receives a 300% bonus, it will still be capped at 90% instead of 100%.